app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java

```diff
@@ -272,10 +273,12 @@ protected void onPostExecute(Tag tag) {
                 if (succeeded) {
                     final String tagId = bytesToHexString(tag.getId());
                     long now = JoH.tsl();
+                    String SensorSn = LibreUtils.decodeSerialNumberKey(tag.getId());
                     boolean checksum_ok = HandleGoodReading(tagId, data, now);
                     if(checksum_ok == false) {
                         Log.e(TAG, "Read data but checksum is wrong");
                     }
+                    PersistentStore.setString("LibreSN", SensorSn);
```

Comment:
why not moving this call within the decode Serial method? like this all collectors can show serial number in system Status.